src/ORM/DataObject.php

```diff
@@ -4186,6 +4187,31 @@ public function mergeRelatedObjects($list, $items)
         return $added;
     }
 
+    /**
+     * Generate a unique key for data object
+     * the unique key uses the @see DataObject::getCacheKeyComponent() extension point so unique key modifiers
+     * such as versioned or fluent are covered
+     * i.e. same data object in different stages or different locales will produce different unique key
+     *
+     * recommended use:
+     * - when you need unique key for caching purposes
+     * - when you need unique id on the front end (for example JavaScript needs to target specific element)
+     *
+     * @return string
+     */
+    public function getUniqueKey(): string
+    {
+        /** @var UniqueKeyInterface $service */
+        $service = UniqueKeyService::singleton();
+        if (!$service instanceof UniqueKeyInterface) {
+            return '';
```

Comment:
Again — I think we should return a random number rather than an empty string if we don't know how to generate a unique key.